packages/CodingStandard/src/Fixer/Property/ArrayPropertyDefaultValueFixer.php

```diff
@@ -70,8 +70,13 @@ private function isArrayPropertyDocComment(Token $token): bool
         return true;
     }
 
-    private function addDefaultValueForArrayProperty(Token $semicolonToken): void
+    private function addDefaultValueForArrayProperty(Tokens $tokens,int $semicolonPosition): void
     {
-        $semicolonToken->setContent(' = [];');
+        $tokens->insertAt($semicolonPosition, new Token(']'));
+        $tokens->insertAt($semicolonPosition, new Token('['));
+        $tokens->insertAt($semicolonPosition, new Token(' '));
```

Comment:
still, it would be a method at `Tokens`, not at `Token`.
originally we had solution that works similar to what you propose internally (it wasn't nice for dev, but it done what you are saying). by removing it we gained ~50% performance boost.

about knowledge - I agree that sth may not be clear, if you are able to improve docs - go ahead !
* backwards - general programming knowledge, not related to PHP CS Fixer at all, neither to PHP only.
* where to find tokens - links PHP list of tokens and our registerec `CT` tokens
* `array vs string` - internal php shit, ask them; `CT` - reasoning is hidden indeed, if interested, go for https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.3.0/src/Tokenizer/TransformerInterface.php#L20-L21